src/ray/gcs/redis_context.cc

```diff
@@ -153,13 +153,15 @@ const std::vector<std::optional<std::string>> &CallbackReply::ReadAsStringArray(
 
 RedisRequestContext::RedisRequestContext(instrumented_io_context &io_service,
                                          RedisCallback callback,
-                                         RedisAsyncContext *context,
+                                         std::shared_ptr<RedisAsyncContext> &&context,
+                                         RedisContext &parent_context,
                                          std::vector<std::string> args)
     : exp_back_off_(RayConfig::instance().redis_retry_base_ms(),
                     RayConfig::instance().redis_retry_multiplier(),
                     RayConfig::instance().redis_retry_max_ms()),
       io_service_(io_service),
-      redis_context_(context),
+      redis_context_(std::move(context)),
+      parent_context_(parent_context),
```

Comment:
Actually do we need to store both, if we have parent_context_, we can get redis_context_ from it right?